examples/water_sampling_hrex.py

```diff
@@ -0,0 +1,355 @@
+
+    print("Bisected Lambda Schedule", lambda_schedule)
+
+    # Second phase: sample initial states determined by bisection using HREX
+    pair_bar_result, frames, boxes, diagnostics = run_sims_hrex(
+        initial_states_hrex, md_params, n_frames_per_iter=n_frames_per_iter, temperature=temperature
+    )
+
+    plots = make_pair_bar_plots(pair_bar_result, temperature, combined_prefix)
+
+    stored_frames = []
+    stored_boxes = []
+    keep_idxs = np.arange(len(initial_states_hrex))
+    for i in keep_idxs:
+        stored_frames.append(frames[i])
+        stored_boxes.append(boxes[i])
+
+    return (
+        SimulationResult(pair_bar_result, plots, stored_frames, stored_boxes, md_params, None, diagnostics),
+        lambda_schedule,
+    )
+
+
+def build_system(host_pdbfile: str, water_ff: str, padding: float):
+    if isinstance(host_pdbfile, str):
+        assert os.path.exists(host_pdbfile)
+        host_pdb = app.PDBFile(host_pdbfile)
+    elif isinstance(host_pdbfile, app.PDBFile):
+        host_pdb = host_pdbfile
+    else:
+        raise TypeError("host_pdbfile must be a string or an openmm PDBFile object")
+
+    host_coords = strip_units(host_pdb.positions)
+
+    box_lengths = np.amax(host_coords, axis=0) - np.amin(host_coords, axis=0)
+
+    box_lengths = box_lengths + padding
+    box = np.eye(3, dtype=np.float64) * box_lengths
+
+    host_ff = app.ForceField(f"{water_ff}.xml")
+    nwa = len(host_coords)
+    solvated_host_system = host_ff.createSystem(
+        host_pdb.topology, nonbondedMethod=app.NoCutoff, constraints=None, rigidWater=False
+    )
+    solvated_host_coords = host_coords
+    solvated_topology = host_pdb.topology
+
+    return solvated_host_system, solvated_host_coords, box, solvated_topology, nwa
+
+
+def get_initial_state(water_pdb, mol, ff, seed, nb_cutoff, lamb):
+    assert nb_cutoff == 1.2  # hardcoded in prepare_host_edge
+
+    # read water system
+    solvent_sys, solvent_conf, solvent_box, solvent_topology, num_water_atoms = build_system(
+        water_pdb, ff.water_ff, padding=0.1
+    )
+    solvent_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+
+    assert num_water_atoms == len(solvent_conf)
+
+    bt = BaseTopology(mol, ff)
+    afe = AbsoluteFreeEnergy(mol, bt)
+    host_config = HostConfig(solvent_sys, solvent_conf, solvent_box, num_water_atoms)
+    potentials, params, combined_masses = afe.prepare_host_edge(ff.get_params(), host_config, lamb)
+
+    ligand_idxs = np.arange(num_water_atoms, num_water_atoms + mol.GetNumAtoms())
+
+    # re-parameterize ligand idxs to do decouple only half of the buckyball
+    nb_params = params[-1]
+
+    final_masses = combined_masses
+    final_conf = np.concatenate([solvent_conf, get_romol_conf(mol)], axis=0)
+
+    component_dim = 4  # q,s,e,w
+    num_atoms = len(final_conf)
+
+    ligand_water_flat_idxs = np.s_[component_dim * num_atoms : 2 * component_dim * num_atoms]  # water
```

Comment:
nit: would it be cleaner to first reshape `ligand_water_flat_idxs` to `(num_nonbonded_potentials, num_atoms, component_dim)`?

Reply:
probably - I copied this code from another branch that I was fairly certain was working, so I didn't want to mess with the indexing here too much